packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -1080,16 +1136,12 @@ export const DatePickerMixin = (subclass) =>
      * @override
      */
     _onEnter(_event) {
-      const oldValue = this.value;
       if (this.opened) {
         // Closing will implicitly select parsed or focused date
         this.close();
       } else {
         this.__commitParsedOrFocusedDate();
       }
-      if (oldValue === this.value) {
```

Comment:
**note:** Now that the validation is the responsibility of the commit logic, we no longer need to trigger it here. This fixes https://github.com/vaadin/web-components/issues/6591.